src/QSymbolicsBase/predefined_fock.jl

```diff
@@ -161,4 +180,42 @@ julia> qsimplify(S*vac, rewriter=qsimplify_fock)
     basis::FockBasis
 end
 SqueezeOp(z::Number) = SqueezeOp(z, inf_fock_basis)
-symbollabel(x::SqueezeOp) = "S($(x.z))"
\ No newline at end of file
+symbollabel(x::SqueezeOp) = "S($(x.z))"
+
+"""Thermal bosonic state in defined Fock basis."""
+@withmetadata struct ThermalState <: AbstractSingleBosonOp
+    photons::Int
+    basis::FockBasis
```

Comment:
another issue that I am only now noticing that should be fixed in the rest of these operators and states is that `basis` should not be a slot in the struct. Only one single basis makes sense here, `inf_fock_basis`. We should be defining the `basis` method, not adding a basis slot. We should track this issue.

Reply:
as with the previous one, no need to fix it in this PR, but we should track it